### ISSUE TITLE
Showing VRs in Infrastructure > Virtual Routers fails due to CloudStackVersion.parse(String) 

### ISSUE THREAD

**Issue by DennisKonrad** [apache/cloudstack#2917]
##### ISSUE TYPE
 * Bug Report

##### COMPONENT NAME
UI
CloudStackVersion

##### CLOUDSTACK VERSION
master as of Oct 22, 2018
newest commit seems to be #2914 

~~~

~~~

##### OS / ENVIRONMENT
Installation of the management server is on CentOS 7.5.1804

##### SUMMARY
Hi,

showing the VRs is not possible in master branch because of the following error:

![image](https://user-images.githubusercontent.com/33484272/47306356-7edb3380-d62c-11e8-8778-f1d8879e4d2f.png)

org.apache.cloudstack.utils.CloudStackVersion.parse(String) passed 0, but requires a value in the format of int.int.int(.int)(-<legacy patch>)

We are currently using the following Version-String: 

![image](https://user-images.githubusercontent.com/33484272/47306414-a7fbc400-d62c-11e8-87e4-b8cb24ea5f1b.png)

4.12.0.0-1540218507

But it seems it is just not set when trying to access the Virtual Routers in Infrastructure.

##### STEPS TO REPRODUCE
Open Infrastructure > Virtual Routers

##### EXPECTED RESULTS
Seeing a list of the vrs.

##### ACTUAL RESULTS
Seeing error message pictured above.

**Comment by rhtyd** [comment]
Checked, could not reproduce this on 4.11 branch. This seems specific to only master branch.

**Comment by khos2ow** [comment]
@DennisKonrad can you check the VR version directly from the machine? `/etc/cloudstack-release`.

The weird thing about your screenshot is that it complains `0` is not of a correct format in which it should have been `4.12.0.0`. Basically what the reformatted code does is strip anything after first occurrence of `-` and process the rest. Which means the value passed to `org.apache.cloudstack.utils.CloudStackVersion.parse()` is `0-<something>`.

**Comment by DennisKonrad** [comment]
Hi @khos2ow,

it took me some time to double check on this. It seems there is a problem with the systemvm's we are building at the moment "cloudstack-release" is set to the following for us:

![image](https://user-images.githubusercontent.com/33484272/47394548-9ea15300-d723-11e8-9afd-dc5703dfe61b.png)

So I think we have to modify something about our SVM-build process so the output of `/etc/cloudstack-release` matches one of the syntaxes in the tests you updated?

Thank you very much for this hint

**Comment by khos2ow** [comment]
Yes it should have the same format as in `int.int.int[.int][-branding][-snapshot|timestamp|build_number]`

**Comment by DennisKonrad** [comment]
@khos2ow your hint led me in the right direction
Many apologies for this. Our jenkins build script had some bug and therefor did not set the version argument for the svm build.sh anymore

But because the error was  so misleading we thought there is a bug in your code. Many apologies for this. Thanks for helping

**Comment by DennisKonrad** [comment]
So now `/etc/cloudstack-release` is back to:

![image](https://user-images.githubusercontent.com/33484272/47395661-d6aa9500-d727-11e8-98c1-e3e848f882a2.png)

This should work fine now. Thank again

**Comment by DennisKonrad** [comment]
One has to really watch out with build.sh to not get it right. There's one zero to much. I have to fix it again but then it will be right

**Comment by DennisKonrad** [comment]
@khos2ow Maybe the version string produced by build.sh without any arguments should also be regarded as a valid version string for a svm so things like that cannot happen?

**Comment by khos2ow** [comment]
No problem at all. There are flags in `build.sh` which you can use to differentiate the incremental builds (I'm AFK now, will check later) but it would be better to produce something like this `4.12.0.0-63` to prevent confusion. (Use `-` instead of `.`)

**Comment by khos2ow** [comment]
Yes any thing which conforms to the type of _version_ with additional optinal branding or timestamp (separated by `-`) will be regarded as valid.

**Comment by DennisKonrad** [comment]
Oh ok, now I fully understand where our problem was coming from.
The build.sh used the BUILD_NUMBER from jenkins without us knowing it and therefor version was set
automatically to the build number in jenkins.

I did not recognize that the magically incrementing number was the same as in jenkins. So this clears it up for me.

Ok, I'm not sure if this automagic link to the jenkins build number is a good thing if it automagically produces systemvm's that will later fail with the error message we got.

Of course it might be a case that happens not so often but if one would like to use the build number he could just call build.sh in jenkins and set the build number as an argument himself.
